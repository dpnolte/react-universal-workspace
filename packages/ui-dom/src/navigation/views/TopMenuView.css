.top-menu-link-text {
  padding-left: 5px;
  padding-top: 3px;
  color: #bbb;
}
.container {
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  display: flex;
  flex-direction: column;
}

.header {
  height: 60px;
}

.content {
  display: flex;
  height: 100%;
}

/* A simple, css only, (some-what) responsive menu */

nav.top-menu-nav {
  position: relative;
  font-size: 13px;
  text-transform: uppercase;
  text-align: center;
  /* Permalink - use to edit and share this gradient: http://colorzilla.com/gradient-editor/#2b2b2b+7,333333+100 */
  background: #2b2b2b; /* Old browsers */
  background: -moz-linear-gradient(
    top,
    #2b2b2b 7%,
    #333333 100%
  ); /* FF3.6-15 */
  background: -webkit-linear-gradient(
    top,
    #2b2b2b 7%,
    #333333 100%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
    to bottom,
    #2b2b2b 7%,
    #333333 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#2b2b2b', endColorstr='#333333',GradientType=0 ); /* IE6-9 */
}

.top-menu-nav a {
  text-decoration: none;
  color: #fff;
  display: block;
}

.top-menu-nav ul {
  list-style: none;
  margin-block-start: 0;
  margin-block-end: 0;
  position: relative;
  text-align: left;
}

.top-menu-nav li {
  float: left;
}

/* clear'n floats */
.top-menu-nav ul:after {
  clear: both;
}

.top-menu-nav ul:before,
.top-menu-nav ul:after {
  content: ' ';
  display: table;
}

/* prime */
.top-menu-nav ul.primary li a {
  padding: 20px 30px;
  border-right: 1px solid #3d3d3d;
  display: flex;
  flex-direction: row;
}

.top-menu-nav ul.primary li a:hover {
  color: #000;
}

/* subs */
.top-menu-nav ul.sub {
  position: absolute;
  z-index: 200;
  box-shadow: 2px 2px 0 #bebebe;
  width: 20%;
  display: none;
}

.top-menu-nav ul.sub li {
  float: none;
  margin: 0;
}

.top-menu-nav ul.sub li a {
  border-bottom: 1px dotted #ccc;
  border-right: none;
  color: #000;
  padding: 15px 30px;
}

.top-menu-nav ul.sub li:last-child a {
  border-bottom: none;
}

.top-menu-nav ul.sub li a:hover {
  color: #000;
  background: #eeeeee;
}

/* sub display*/
.top-menu-nav ul.primary li:hover ul {
  display: block;
  background: #fff;
  margin-top: -3px;
}

/* keeps the tab background white */
.top-menu-nav ul.primary li:hover a {
  background: #fff;
  color: #666;
  text-shadow: none;
}

.top-menu-nav ul.primary li:hover > a .top-menu-link-text {
  color: #999;
}
