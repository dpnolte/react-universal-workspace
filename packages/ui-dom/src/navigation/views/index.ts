export * from './TopMenuView'
export * from './SinglePageView'
