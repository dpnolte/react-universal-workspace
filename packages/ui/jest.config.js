// use web by default
// eslint-disable-next-line @typescript-eslint/no-var-requires
const webConfig = require('./test/jest.config.dom')

module.exports = webConfig
