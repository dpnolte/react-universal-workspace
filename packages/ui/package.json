{
  "name": "@mono/ui",
  "version": "0.0.1",
  "license": "MIT",
  "private": true,
  "scripts": {
    "check-types": "tsc --project ./tsconfig.json"
  },
  "main": "src/index.ts",
  "dependencies": {
    "@mono/core": "0.0.1",
    "react": "16.8.3",
    "react-native": "0.59.5",
    "react-native-svg": "^9.4.0",
    "@react-navigation/core": "^3.4.1",
    "react-native-vector-icons": "^6.4.2",
    "mobx": "^5.9.4",
    "mobx-react-lite": "^1.3.2"
  },
  "devDependencies": {
    "@types/jest": "^24.0.11",
    "@types/react": "^16.8.14",
    "@types/react-native": "^0.57.46",
    "jest": "^24.7.1",
    "typescript": "^3.4.4",
    "@mono/scripts": "0.0.1",
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/plugin-proposal-object-rest-spread": "^7.4.4",
    "@babel/plugin-syntax-dynamic-import": "^7.2.0",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-typescript": "^7.3.3",
    "babel-eslint": "^10.0.1",
    "babel-jest": "^24.0.0",
    "@babel/preset-react": "^7.0.0",
    "core-js": "2.6.5"
  }
}
