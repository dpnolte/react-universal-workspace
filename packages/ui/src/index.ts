// here we can have ui that are being shared among fronte-end touchpoints
// in this package we use react
import * as ScreensImport from './screens'
import * as ElementsImport from './elements'
import * as ContextsImport from './contexts'

export const Screens = ScreensImport
export const Elements = ElementsImport
export const Contexts = ContextsImport
