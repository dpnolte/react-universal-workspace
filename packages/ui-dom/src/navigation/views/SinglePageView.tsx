/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react'
import {
  SceneView,
  NavigationScreenProp,
  NavigationRoute,
  NavigationDescriptor,
} from '@react-navigation/core'
import './TopMenuView.css'

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface INavigationParams {}
interface IProps {
  descriptors: { [key: string]: NavigationDescriptor }
  navigation: NavigationScreenProp<
    NavigationRoute<INavigationParams>,
    INavigationParams
  >
  navigationConfig: {}
}

export const SinglePageView = (props: IProps) => {
  const { descriptors, navigation } = props
  const activeKey = navigation.state.routes[navigation.state.index].key
  const descriptor = descriptors[activeKey]
  return (
    <SceneView
      navigation={descriptor.navigation}
      component={descriptor.getComponent()}
    />
  )
}
