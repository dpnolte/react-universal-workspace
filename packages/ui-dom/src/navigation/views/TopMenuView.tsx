/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react'
import {
  SceneView,
  NavigationScreenProp,
  NavigationRoute,
  NavigationDescriptor,
} from '@react-navigation/core'
import { Link } from '@react-navigation/web'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
import './TopMenuView.css'

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface INavigationParams {}
interface IProps {
  descriptors: { [key: string]: NavigationDescriptor }
  navigation: NavigationScreenProp<
    NavigationRoute<INavigationParams>,
    INavigationParams
  >
  navigationConfig: {}
}

export const TopMenuView = (props: IProps) => {
  const { descriptors, navigation } = props
  const activeKey = navigation.state.routes[navigation.state.index].key
  const descriptor = descriptors[activeKey]
  return (
    <div className="container">
      <header className="header">
        <nav className="top-menu-nav">
          <ul className="primary">
            <li>
              <Link
                routeName="Home"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.HomeLink,
                }}
              >
                <Icon name="home" size={20} color="#bbb" />
                <span className="top-menu-link-text">Home</span>
              </Link>
            </li>
            <li>
              <Link
                routeName="Docs"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.DocsLink,
                }}
              >
                <Icon name="file-document-outline" size={20} color="#bbb" />
                <span className="top-menu-link-text">Documents</span>
              </Link>
              {/* <ul className="sub">
                <li>
                  <Link
                    routeName="Docs"
                    navigation={navigation}
                    anchorProps={{
                      'data-testid': TopMenuViewTestIds.DocsLink,
                    }}
                  >
                    Doc 1
                  </Link>
                </li>
                  </ul> */}
            </li>
            <li>
              <Link
                routeName="TodoList"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.TodoListLink,
                }}
              >
                <Icon name="format-list-checkbox" size={20} color="#bbb" />
                <span className="top-menu-link-text">Todo List</span>
              </Link>
            </li>
            <li>
              <Link
                routeName="About"
                navigation={navigation}
                anchorProps={{
                  'data-testid': TopMenuViewTestIds.AboutLink,
                }}
              >
                <Icon name="vector-arrange-above" size={20} color="#bbb" />
                <span className="top-menu-link-text">About</span>
              </Link>
            </li>
          </ul>
        </nav>
      </header>
      <section className="content">
        <SceneView
          navigation={descriptor.navigation}
          component={descriptor.getComponent()}
        />
      </section>
    </div>
  )
}

export const TopMenuViewTestIds = {
  HomeLink: 'top-menu-home-link',
  DocsLink: 'top-menu-docs-link',
  AboutLink: 'top-menu-about-link',
  TodoListLink: 'top-menu-todo-list-link',
}
