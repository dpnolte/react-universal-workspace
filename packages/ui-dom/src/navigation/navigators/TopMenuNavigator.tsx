import { Screens } from '@mono/ui'
import { createNavigator, SwitchRouter } from '@react-navigation/core'
import { TopMenuView } from '../views'

export const RouteNames = {
  Home: 'Home',
  About: 'About',
  Docs: 'Docs',
  TodoList: 'TodoList',
}

export const TopMenuNavigator = createNavigator(
  TopMenuView,
  SwitchRouter({
    [RouteNames.Home]: {
      screen: Screens.Home,
      navigationOptions: {
        title: 'Home',
      },
    },
    [RouteNames.About]: {
      screen: Screens.About,
      navigationOptions: {
        title: 'About',
      },
    },
    [RouteNames.Docs]: {
      screen: Screens.Docs,
      navigationOptions: {
        title: 'Docs',
      },
    },
    [RouteNames.TodoList]: {
      screen: Screens.TodoList,
      navigationOptions: {
        title: 'Todo List',
      },
    },
  }),
  {}
)
