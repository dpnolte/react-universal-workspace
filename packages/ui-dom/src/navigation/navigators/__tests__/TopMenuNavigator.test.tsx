import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent, cleanup } from 'react-testing-library'
import { Screens } from '@mono/ui'
import { createBrowserApp } from '@react-navigation/web'
import { TopMenuNavigator } from '../TopMenuNavigator'
import { TopMenuViewTestIds } from '../../views'
import 'jest-dom/extend-expect'

jest.mock('history', () => ({
  createBrowserHistory: () => ({
    push: () => {},
    listen: () => {},
    location: {
      pathname: '/Home',
    },
  }),
}))

document.title = 'Empty Title'

describe('TopMenuNavigator test suite', () => {
  let BrowserApp: React.ComponentType
  beforeEach(() => {
    BrowserApp = (createBrowserApp(
      TopMenuNavigator
    ) as unknown) as React.ComponentType
  })
  afterEach(() => {
    cleanup()
  })
  it('should match snapshot', () => {
    const rendered = renderer.create(<BrowserApp />).toJSON()

    expect(rendered).toMatchSnapshot()
  })
  it('should go to docs screen and then go to home', () => {
    const { getByTestId, queryByTestId, queryAllByTestId } = render(
      <BrowserApp />
    )
    const homeScreenBefore = queryByTestId(Screens.HomeScreenTestId)
    expect(homeScreenBefore).toBeInTheDocument()

    const docLinks = queryAllByTestId(TopMenuViewTestIds.DocsLink)
    expect(docLinks).toHaveLength(2)
    fireEvent.click(docLinks[0])
    const docsScreen = queryByTestId(Screens.DocsScreenTestId)
    expect(docsScreen).toBeInTheDocument()
    fireEvent.click(getByTestId(TopMenuViewTestIds.HomeLink))
    const homeScreenAfter = queryByTestId(Screens.HomeScreenTestId)
    expect(homeScreenAfter).toBeInTheDocument()
  })
  it('should go to about screen and then go to home', () => {
    const { getByTestId, queryByTestId } = render(<BrowserApp />)

    const homeScreenBefore = queryByTestId(Screens.HomeScreenTestId)
    expect(homeScreenBefore).toBeInTheDocument()
    fireEvent.click(getByTestId(TopMenuViewTestIds.AboutLink))

    const aboutScreen = queryByTestId(Screens.AboutScreenTestId)
    expect(aboutScreen).toBeInTheDocument()

    fireEvent.click(getByTestId(TopMenuViewTestIds.HomeLink))
    const homeScreenAfter = queryByTestId(Screens.HomeScreenTestId)
    expect(homeScreenAfter).toBeInTheDocument()
  })
})
