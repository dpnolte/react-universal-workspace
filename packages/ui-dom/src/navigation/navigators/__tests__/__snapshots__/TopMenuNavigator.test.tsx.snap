// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenuNavigator test suite should match snapshot 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <nav
      className="top-menu-nav"
    >
      <ul
        className="primary"
      >
        <li>
          <a
            data-testid="top-menu-home-link"
            href="/Home"
            onClick={[Function]}
          >
            <div
              className="css-text-76zvg2"
              dir="auto"
              style={
                Object {
                  "color": "rgba(187,187,187,1.00)",
                  "fontFamily": "MaterialCommunityIcons",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                }
              }
            >
              
            </div>
            <span
              className="top-menu-link-text"
            >
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            data-testid="top-menu-docs-link"
            href="/Docs"
            onClick={[Function]}
          >
            <div
              className="css-text-76zvg2"
              dir="auto"
              style={
                Object {
                  "color": "rgba(187,187,187,1.00)",
                  "fontFamily": "MaterialCommunityIcons",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                }
              }
            >
              吝
            </div>
            <span
              className="top-menu-link-text"
            >
              Documents
            </span>
          </a>
          <ul
            className="sub"
          >
            <li>
              <a
                data-testid="top-menu-docs-link"
                href="/Docs"
                onClick={[Function]}
              >
                Doc 1
              </a>
            </li>
          </ul>
        </li>
        <li>
          <a
            data-testid="top-menu-about-link"
            href="/About"
            onClick={[Function]}
          >
            <div
              className="css-text-76zvg2"
              dir="auto"
              style={
                Object {
                  "color": "rgba(187,187,187,1.00)",
                  "fontFamily": "MaterialCommunityIcons",
                  "fontSize": "20px",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                }
              }
            >
              
            </div>
            <span
              className="top-menu-link-text"
            >
              About
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <section
    className="content"
  >
    <main
      className="css-view-1dbjc4n"
      data-testid="home-screen"
      role="main"
      style={
        Object {
          "backgroundColor": "rgba(34,34,34,1.00)",
          "height": "100%",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "overflowX": "hidden",
          "overflowY": "hidden",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(255,255,255,1.00)",
            "fontSize": "14px",
            "lineHeight": "21px",
            "marginBottom": "20px",
            "marginTop": "20px",
            "textAlign": "center",
          }
        }
      >
        Welcome to this mono repo
      </div>
      <h1
        className="css-reset-4rbku5 css-text-76zvg2"
        dir="auto"
        role="heading"
        style={
          Object {
            "backgroundColor": "rgba(34,34,34,0.50)",
            "color": "rgba(255,255,255,1.00)",
            "fontSize": "32px",
            "fontWeight": "bold",
            "marginBottom": "20px",
            "marginTop": "20px",
            "textAlign": "center",
            "zIndex": 999,
          }
        }
      >
        ONE REPO TO RULE THEM ALL
      </h1>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxPack": "center",
            "WebkitJustifyContent": "center",
            "WebkitTransform": "rotate(0deg)",
            "alignItems": "center",
            "bottom": "0px",
            "justifyContent": "center",
            "left": "0px",
            "msFlexAlign": "center",
            "msFlexPack": "center",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "transform": "rotate(0deg)",
          }
        }
      >
        <svg
          height="100%"
          viewBox="0 0 841.9 595.3"
          width="100%"
        >
          <g
            fill="#61DAFB"
          >
            <path
              d="M666.3 296.5c0-32.5-40.7-63.3-103.1-82.4 14.4-63.6 8-114.2-20.2-130.4-6.5-3.8-14.1-5.6-22.4-5.6v22.3c4.6 0 8.3.9 11.4 2.6 13.6 7.8 19.5 37.5 14.9 75.7-1.1 9.4-2.9 19.3-5.1 29.4-19.6-4.8-41-8.5-63.5-10.9-13.5-18.5-27.5-35.3-41.6-50 32.6-30.3 63.2-46.9 84-46.9V78c-27.5 0-63.5 19.6-99.9 53.6-36.4-33.8-72.4-53.2-99.9-53.2v22.3c20.7 0 51.4 16.5 84 46.6-14 14.7-28 31.4-41.3 49.9-22.6 2.4-44 6.1-63.6 11-2.3-10-4-19.7-5.2-29-4.7-38.2 1.1-67.9 14.6-75.8 3-1.8 6.9-2.6 11.5-2.6V78.5c-8.4 0-16 1.8-22.6 5.6-28.1 16.2-34.4 66.7-19.9 130.1-62.2 19.2-102.7 49.9-102.7 82.3 0 32.5 40.7 63.3 103.1 82.4-14.4 63.6-8 114.2 20.2 130.4 6.5 3.8 14.1 5.6 22.5 5.6 27.5 0 63.5-19.6 99.9-53.6 36.4 33.8 72.4 53.2 99.9 53.2 8.4 0 16-1.8 22.6-5.6 28.1-16.2 34.4-66.7 19.9-130.1 62-19.1 102.5-49.9 102.5-82.3zm-130.2-66.7c-3.7 12.9-8.3 26.2-13.5 39.5-4.1-8-8.4-16-13.1-24-4.6-8-9.5-15.8-14.4-23.4 14.2 2.1 27.9 4.7 41 7.9zm-45.8 106.5c-7.8 13.5-15.8 26.3-24.1 38.2-14.9 1.3-30 2-45.2 2-15.1 0-30.2-.7-45-1.9-8.3-11.9-16.4-24.6-24.2-38-7.6-13.1-14.5-26.4-20.8-39.8 6.2-13.4 13.2-26.8 20.7-39.9 7.8-13.5 15.8-26.3 24.1-38.2 14.9-1.3 30-2 45.2-2 15.1 0 30.2.7 45 1.9 8.3 11.9 16.4 24.6 24.2 38 7.6 13.1 14.5 26.4 20.8 39.8-6.3 13.4-13.2 26.8-20.7 39.9zm32.3-13c5.4 13.4 10 26.8 13.8 39.8-13.1 3.2-26.9 5.9-41.2 8 4.9-7.7 9.8-15.6 14.4-23.7 4.6-8 8.9-16.1 13-24.1zM421.2 430c-9.3-9.6-18.6-20.3-27.8-32 9 .4 18.2.7 27.5.7 9.4 0 18.7-.2 27.8-.7-9 11.7-18.3 22.4-27.5 32zm-74.4-58.9c-14.2-2.1-27.9-4.7-41-7.9 3.7-12.9 8.3-26.2 13.5-39.5 4.1 8 8.4 16 13.1 24 4.7 8 9.5 15.8 14.4 23.4zM420.7 163c9.3 9.6 18.6 20.3 27.8 32-9-.4-18.2-.7-27.5-.7-9.4 0-18.7.2-27.8.7 9-11.7 18.3-22.4 27.5-32zm-74 58.9c-4.9 7.7-9.8 15.6-14.4 23.7-4.6 8-8.9 16-13 24-5.4-13.4-10-26.8-13.8-39.8 13.1-3.1 26.9-5.8 41.2-7.9zm-90.5 125.2c-35.4-15.1-58.3-34.9-58.3-50.6 0-15.7 22.9-35.6 58.3-50.6 8.6-3.7 18-7 27.7-10.1 5.7 19.6 13.2 40 22.5 60.9-9.2 20.8-16.6 41.1-22.2 60.6-9.9-3.1-19.3-6.5-28-10.2zM310 490c-13.6-7.8-19.5-37.5-14.9-75.7 1.1-9.4 2.9-19.3 5.1-29.4 19.6 4.8 41 8.5 63.5 10.9 13.5 18.5 27.5 35.3 41.6 50-32.6 30.3-63.2 46.9-84 46.9-4.5-.1-8.3-1-11.3-2.7zm237.2-76.2c4.7 38.2-1.1 67.9-14.6 75.8-3 1.8-6.9 2.6-11.5 2.6-20.7 0-51.4-16.5-84-46.6 14-14.7 28-31.4 41.3-49.9 22.6-2.4 44-6.1 63.6-11 2.3 10.1 4.1 19.8 5.2 29.1zm38.5-66.7c-8.6 3.7-18 7-27.7 10.1-5.7-19.6-13.2-40-22.5-60.9 9.2-20.8 16.6-41.1 22.2-60.6 9.9 3.1 19.3 6.5 28.1 10.2 35.4 15.1 58.3 34.9 58.3 50.6-.1 15.7-23 35.6-58.4 50.6zM320.8 78.4z"
            />
            <circle
              cx="420.9"
              cy="296.5"
              r="45.7"
            />
            <path
              d="M520.5 78.1z"
            />
          </g>
        </svg>
      </div>
    </main>
  </section>
</div>
`;
